tests/test_integration/test_code_generation.py

```diff
@@ -179,3 +179,43 @@ def stencil(field_in: gtscript.Field[np.float_], field_out: gtscript.Field[np.fl
 
     np.testing.assert_allclose(field_out.view(np.ndarray)[:, :, 0:-1], 3)
     np.testing.assert_allclose(field_out.view(np.ndarray)[:, :, -1], 2)
+
+
+@pytest.mark.parametrize("backend", CPU_BACKENDS)
+def test_lower_dimensional_inputs(backend):
+    @gtscript.stencil(backend=backend)
+    def stencil(
+        field_3d: gtscript.Field[np.float_, gtscript.IJK],
+        field_2d: gtscript.Field[np.float_, gtscript.IJ],
+        field_1d: gtscript.Field[np.float_, gtscript.K],
+    ):
+        with computation(PARALLEL):
+            with interval(0, 1):
+                field_2d = field_1d[1] + field_3d[0, 1, 0]
```

Comment:
This should actually be invalid, since for any larger interval, this is a  data race.

Reply:
Ah yes, this should be a `FORWARD` computation.